RfidShelf/ShelfRfid.h

```diff
@@ -8,6 +8,15 @@
 #include <MFRC522.h>
 #include <SdFat.h>
 
+// this object stores nfc tag data
+struct nfcTagObject {
+  char folder[17] = {'\0'};
+  uint8_t volume;
+  bool repeat;
+  bool shuffle;
+  bool stopOnRemove;
```

Comment:
The volume is already a per-card setting, but I really have current use cases for the 3 new options.

I have two children. 4 yo and 16 months old. They use the same box for different cards.
The 4 yo uses it to hear audio books before going to bed. So I need the repeat to be off, otherwise he never fell asleep.
The 16 months old has cards with animals on it and hears the sounds of this animal. It's very easy for him to put the card on the box, but it's extremely difficult to explain to him that this card has to stay on the box. So I need the stopOnRemove feature to be disabled.
Shuffle mode is self-explaining I think. In other projects it is called "party mode". I do want to hear audio books in the correct order, but I want to play party music or other music compilation in random order.

I think that it could be overridable in the interface, but for me it is not an option to change the mode every time the user changes. Starting with the point that I am not the only baby sitter and the others have very little technical background and no clue how to open the UI.

Reply:
Volume and shuffle are usefull per card without a doubt. For repeat it might already be questionable, but still makes kind of sense.
For the "stopOnRemove" I think it's different, but of course your 2-user scenario makes total sense (just nothing I would have considered).

I wonder if it makes sense to have some equivalent for NULL on the card config options to keep the cards as "stupid" as possible, but enable them to override the shelfs default when it's needed, like in your case. (NULL would be "Use whatever is currently enabled for this shelf")

Reply:
A NULL is difficult, because I use a bit mask to save the values using just one byte. And  a bit mask does not allow something different than 0 or 1 and AFAIK the standard bool value in C has no NULL value as well. You could leave the whole option block ( = config option byte), but that would leave all 3 options out.

It could be combined with playingByCard flag. Currently shuffle is overridable by the general setting.

On the other hand the standard values are the same as the default setting at the moment (repeat = true, shuffle = false, stopOnRemove = false)

Reply:
My idea was more motivated by the plan to have a few more options configurable per shelf (stored in ESPs equivalent of the EEPROM). If we created cards like this now, there will never be a chance to tell if the values on this card were written on purpose or not (= now automatic upgrade of those cards)
On the other hand we have enough space in the config sector to use on byte per config flag to allow a NULL-equivalent, so I think a more future proof approach would be nice. I think this pull request is fine for now, but I think I want to have the nullable fields before releasing the next version to avoid having such cards "out in the field"